pkg/components/internal/testutil/jsonpath.go

```diff
@@ -38,6 +43,60 @@ func unstructredObj(t *testing.T, yamlObj string) *unstructured.Unstructured {
 	return u
 }
 
+// removeYAMLComments converts YAML to JSON and back again, this removes the comments in the YAML
+// and any extra whitespaces spaces.
+func removeYAMLComments(t *testing.T, yamlObj []byte) []byte {
+	jsonObj, err := yamlconv.YAMLToJSON(yamlObj)
+	if err != nil {
+		t.Fatalf("Converting YAML to JSON: %v", err)
+	}
+
+	yamlObj, err = yamlconv.JSONToYAML(jsonObj)
+	if err != nil {
+		t.Fatalf("Converting JSON to YAML: %v", err)
+	}
+
+	return yamlObj
+}
+
+// splitYAMLDocs converts a YAML string with multiple YAML docs separated by `---` into unique
+// objects and returns those objects as a map.
+func splitYAMLDocs(t *testing.T, yamlObj string) map[ObjectMetadata]string {
```

Comment:
We already have `pkg/components/util.SplitManifests`. I think we should only have one logic for splitting manfiests.

Reply:
`pkg/components/util.SplitManifests` does more than just identify the kind, name and version. It also unpacks a list of objects. The consolidation of both can be worked on here: https://github.com/kinvolk/lokomotive/issues/998.

Reply:
We can refactor it, as I strongly believe we shouldn't implement this logic twice.

Reply:
Yep #998 is for that.

Reply:
Then I think we should create an initial replacement for `pkg/components/util.SplitManifests` in some place other than `pkg/components/internal/testutil`, so it can be imported where we use `pkg/components/util.SplitManifests`.

Reply:
Done.